src/org/opendatakit/briefcase/util/FormCache.java

```diff
@@ -122,8 +121,9 @@ public void update() {
           }
         }
       });
-      hashByPath.keySet().stream().filter(p -> !scannedFiles.contains(p)).collect(toList()).forEach(hashByPath::remove);
-      formDefByPath.keySet().stream().filter(p -> !scannedFiles.contains(p)).collect(toList()).forEach(formDefByPath::remove);
+      // Warning: Remove map entries by mutating the key set works because the key set is a view on the map
+      hashByPath.keySet().removeIf(key -> !scannedFiles.contains(key));
+      formDefByPath.keySet().removeIf(key -> !scannedFiles.contains(key));
```

Comment:
Somehow I knew you would reject my suggestion to factor out the common predicate. We’re getting to know each other! :-)